pkg/function/backups_info.go

```diff
@@ -0,0 +1,170 @@
+	"github.com/kanisterio/kanister/pkg/format"
+	"github.com/kanisterio/kanister/pkg/kube"
+	"github.com/kanisterio/kanister/pkg/param"
+	"github.com/kanisterio/kanister/pkg/restic"
+)
+
+const (
+	// DescribeBackupsNamespaceArg provides the namespace
+	DescribeBackupsNamespaceArg = "namespace"
+	// DescribeBackupsArtifactPrefixArg provides the path to restore backed up data
+	DescribeBackupsArtifactPrefixArg = "backupArtifactPrefix"
+	// DescribeBackupsEncryptionKeyArg provides the encryption key to be used for deletes
+	DescribeBackupsEncryptionKeyArg = "encryptionKey"
+	// DescribeBackupsPodOverrideArg contains pod specs to override default pod specs
+	DescribeBackupsPodOverrideArg    = "podOverride"
+	DescribeBackupsJobPrefix         = "get-backups-info-"
+	DescribeBackupsFileCount         = "fileCount"
+	DescribeBackupsSize              = "size"
+	DescribeBackupsSnapshotIDs       = "snapshotIDs"
+	DescribeBackupsPasswordIncorrect = "passwordIncorrect"
+	DescribeBackupsRepoUnavailable   = "repoUnavailable"
+	RawDataStatsMode                 = "raw-data"
+)
+
+func init() {
+	kanister.Register(&DescribeBackupsFunc{})
+}
+
+var _ kanister.Func = (*DescribeBackupsFunc)(nil)
+
+type DescribeBackupsFunc struct{}
+
+func (*DescribeBackupsFunc) Name() string {
+	return "DescribeBackups"
+}
+
+func describeBackups(ctx context.Context, cli kubernetes.Interface, tp param.TemplateParams, namespace, encryptionKey, targetPaths, jobPrefix string, podOverride sp.JSONMap) (map[string]interface{}, error) {
+	options := &kube.PodOptions{
+		Namespace:    namespace,
+		GenerateName: jobPrefix,
+		Image:        kanisterToolsImage,
+		Command:      []string{"sh", "-c", "tail -f /dev/null"},
+		PodOverride:  podOverride,
+	}
+	pr := kube.NewPodRunner(cli, options)
+	podFunc := describeBackupsPodFunc(cli, tp, namespace, encryptionKey, targetPaths)
+	return pr.Run(ctx, podFunc)
+}
+
+func describeBackupsPodFunc(cli kubernetes.Interface, tp param.TemplateParams, namespace, encryptionKey, targetPath string) func(ctx context.Context, pod *v1.Pod) (map[string]interface{}, error) {
+	return func(ctx context.Context, pod *v1.Pod) (map[string]interface{}, error) {
+		// Wait for pod to reach running state
+		if err := kube.WaitForPodReady(ctx, cli, pod.Namespace, pod.Name); err != nil {
+			return nil, errors.Wrapf(err, "Failed while waiting for Pod %s to be ready", pod.Name)
+		}
+		pw, err := getPodWriter(cli, ctx, pod.Namespace, pod.Name, pod.Spec.Containers[0].Name, tp.Profile)
+		if err != nil {
+			return nil, err
+		}
+		defer cleanUpCredsFile(ctx, pw, pod.Namespace, pod.Name, pod.Spec.Containers[0].Name)
+		snapshotIDs, err := restic.GetSnapshotIDs(tp.Profile, cli, targetPath, encryptionKey, namespace, pod.Name, pod.Spec.Containers[0].Name)
+		switch {
+		case err == nil:
+			break
+		case strings.Contains(err.Error(), restic.PasswordIncorrect):
+			return map[string]interface{}{
+					DescribeBackupsSnapshotIDs:       nil,
```

Comment:
Should these keys be set with nil values or omitted completely?

Reply:
I think we can omit them.